### ISSUE TITLE
[KEv2]: EKS2 if rancher cannot communicate with kube api thread gets stuck

### ISSUE THREAD

**Issue by rmweir** [rancher/rancher#28182]
**What kind of request is this (question/bug/enhancement/feature request):**
bug

**Steps to reproduce (least amount of steps as possible):**
1. create a cluster with only public access and access source endpoints of ["0.0.0.0/16"]

**Result:**
cluster will be stuck in waiting because there is no service account (makes sense), and not updatable because there is no timeout on the client causing the thread to be locked waiting on a Version() call in generate service account logic. Another issue is once this is resolved the endpoints will not update because the wrong value is provided to aws when the endpoints are to use "0.0.0.0/0".

**Expected:**
cluster stuck in waiting initially makes sense, the agent deployer cannot contact the cluster. Should be able to remove the endpoint so access returns to 0.0.0.0/0 and cluster can go to active.

**Comment by rmweir** [comment]
I was mistaken, the timeout is 1 minutes by default and I mistook the longer timeout as being stuck.
